top.chidao0000.Watermelon.yml

```diff
@@ -0,0 +1,61 @@
+id: top.chidao0000.Watermelon
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: watermelon
+modules:
+  - name: watermelon
+    buildsystem: simple
+    build-commands:
+      - install -Dm 755 -t /app/bin/ watermelon
+      - install -Dm 644 top.chidao0000.Watermelon.png /app/share/icons/hicolor/512x512/apps/top.chidao0000.Watermelon.png
+      - install -Dm 644 top.chidao0000.Watermelon.desktop -t /app/share/applications
+
+      # 解压运行环境文件
+      - ls -al
+      - mkdir -p /app/eda
+      - unzip ./eda.zip -d /app/eda/
+      - rm ./eda.zip
+
+      # 解压文件
+      - mkdir -p /app/eda/resources/app
+      - tar -zxvf ./pro-editor-client.tgz -C /app/eda/resources/app
+      - rm ./pro-editor-client.tgz
+
+    sources:
+      - type: script
+        commands:
+          - exec /app/eda/electron --no-sandbox "$@"
+        dest-filename: watermelon
+
+      # 从远端配置下载文件
+      - type: file
+        only-arches: [x86_64]
+        url: https://registry.npmmirror.com/-/binary/electron/33.2.1/electron-v33.2.1-linux-x64.zip
+        sha256: 3b66a507ece29684a105b0f9609db50c88235333ddad696d01eefebefd4b1aab
+        dest-filename: eda.zip
+
+      - type: file
+        only-arches: [aarch64]
+        url: https://registry.npmmirror.com/-/binary/electron/33.2.1/electron-v33.2.1-linux-arm64.zip
+        sha256: 61ef1182fcbcaaf0b9036243216747eb9be186491b79aa6576b482212a928d2e
+        dest-filename: eda.zip
+
+      - type: file
+        url: https://modules.lceda.cn/client/2.2.33.11.a0de9e38.a2293c/pro-editor-client-2.2.33.11.a0de9e38.a2293c.tgz
+        sha256: 8a81dfe47ac8c5c25fc322dc0868d9c4bdc744064c31fc8e1cf6ae0a2decd753
+        dest-filename: pro-editor-client.tgz
+
+      - type: file
+        path: top.chidao0000.Watermelon.png
+
+      - type: file
+        path: top.chidao0000.Watermelon.desktop
+
+
+finish-args:
+  - --share=network
+  - --socket=x11
+  - --socket=wayland
```

Comment:
it either only `x11` or `wayland` + `fallback-x11`. Not both.

Also `--share=ipc` is missing.